modules/oe_whitelabel_search/src/Plugin/Block/SearchBlock.php

```diff
@@ -0,0 +1,211 @@
+      '#required' => TRUE,
+    ];
+    $form['input_label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Input Label'),
+      '#description' => $this->t('A label text for the search input.'),
+      '#default_value' => $config['input']['label'],
+    ];
+    $form['input_placeholder'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Input placeholder text'),
+      '#default_value' => $config['input']['placeholder'],
+    ];
+    $form['button_label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button label'),
+      '#description' => $this->t('Label text that should appear inside the button.'),
+      '#default_value' => $config['button']['label'],
+    ];
+    $form['button_type'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button type'),
+      '#description' => $this->t('Ex: submit, button.'),
+      '#default_value' => $config['button']['type'],
+    ];
+    $form['button_icon_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button icon name'),
+      '#description' => $this->t('Ex: search.'),
+      '#default_value' => $config['button']['icon']['name'],
+    ];
+    $form['button_icon_position'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Button icon position'),
+      '#description' => $this->t('The position of the icon inside the button.'),
+      '#default_value' => $config['button']['icon']['position'],
+    ];
+    $form['enable_autocomplete'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable autocomplete'),
+      '#default_value' => $config['view_options']['enable_autocomplete'] ?? 0,
+    ];
+    $form['view_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View id'),
+      '#description' => $this->t('The view id (referenced as #search_id in the form).'),
+      '#default_value' => $config['view_options']['id'] ?? '',
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[enable_autocomplete]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+    $form['view_display'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('View display'),
+      '#description' => $this->t('The view display.'),
+      '#default_value' => $config['view_options']['display'] ?? '',
+      '#states' => [
+        'visible' => [
+          ':input[name="settings[enable_autocomplete]"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state): void {
+    $this->setConfigurationValue('form', [
+      'action' => $form_state->getValue('form_action'),
+      'classes' => $form_state->getValue('form_classes'),
+    ]);
+    $this->setConfigurationValue('input', [
+      'name' => $form_state->getValue('input_name'),
+      'label' => $form_state->getValue('input_label'),
+      'classes' => $form_state->getValue('input_classes'),
```

Comment:
Where is coming from the input_classes. In the files the input_placeholder is found in different places, but the input_classes only is here. Could you clarify, please?